swan-lake/integration-tutorials/sending-a-message-to-a-service.md

```diff
@@ -264,14 +276,64 @@ time = 2023-08-15T13:01:34.022+05:30 level = INFO module = integration_tutorials
 
 Let's test the use case by writing a test case that sends a request to the service and validates the payload for a successful request. Testing is enabled by the [Ballerina test framework](https://ballerina.io/learn/test-ballerina-code/test-services-and-clients/).
 
-1. Introduce the tests in a `.bal` file within a directory named `tests` in the package. Import the `ballerina/test` module to use the Ballerina test framework and the `ballerina/http` module to use an `http:Client` client object to send requests to the implemented service and mock the backend service.
+1. Change `queryDoctorEP` initialization in the source code to use a separate function that can be mocked.
+
+    Replace
+
+    ```ballerina
+    final http:Client queryDoctorEP = check new (healthcareBackend);
+    ```
+
+    with
+
+    ```ballerina
+    final http:Client queryDoctorEP = check initializeHttpClient();
+
+    function initializeHttpClient() returns http:Client|error => new (healthcareBackend);
+    ```
+
+2. Introduce the tests in a `.bal` file within a directory named `tests` in the package. Import the `ballerina/test` module to use the Ballerina test framework and the `ballerina/http` module to use an `http:Client` client object to send requests to the implemented service and mock the backend service.
 
     ```ballerina
     import ballerina/http;
     import ballerina/test;
     ```
```

Comment:
I think we can remove this step like we did for the source.

Reply:
If we remove this, the user may get errors if he copies just the code snippets. One thing we can do here is give the full test code and explain it one by one. WDYT?

Reply:
Similar concern for the source if they don't use the diagram view, right? I think we can just remove.